android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/MessageCenter.java

```diff
@@ -0,0 +1,77 @@
+package com.shopify.unity.buy;
+
+import android.os.Handler;
+import android.os.HandlerThread;
+
+import com.unity3d.player.UnityPlayer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class MessageCenter {
+    private static final HandlerThread handlerThread = new HandlerThread("UnityMessageThread");
+    private static final Handler messageHandler = new Handler(handlerThread.getLooper());
+
+    private static Map<String, MessageCallbacks> callbacksInWaiting = new HashMap<>();
+
+    private MessageCenter() { }
+
+    // Fire and forget sending of messages.
+    static void sendMessageTo(final UnityMessage msg, final MessageReceiver receiver) {
+        messageHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                UnityPlayer.UnitySendMessage(receiver.unityDelegateObjectName, receiver.method.name, msg.toJsonString());
+            }
+        });
+    }
+
+    // Send message with callbacks to invoke when complete.
+    static void sendMessageTo(final UnityMessage msg, final MessageReceiver receiver, final MessageCallbacks callbacks) {
+        messageHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                callbacksInWaiting.put(msg.identifier, callbacks);
+                UnityPlayer.UnitySendMessage(receiver.unityDelegateObjectName, receiver.method.name, msg.toJsonString());
+            }
+        });
+    }
+
+    @SuppressWarnings("unused")
+    public static void onUnityResponse(final String result) {
+        messageHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                UnityMessage msg = UnityMessage.fromUnity(result);
+                MessageCallbacks callbacks = callbacksInWaiting.remove(msg.identifier);
+                callbacks.onResponse(msg.content);
+            }
+        });
+    }
+
+    static class MessageReceiver {
```

Comment:
This part is confusing, does it represent the call that must be performed on Unity side?
If yes, I would suggest to give a better name for this like: `UnityMessageReceiver`, what do you think?